Chore: refactor preemployment submission

C-Ivan-Bandilla opened the pull request:
Refactor the preemployment submission to use alpine js
- Didnt use [dasundev/livewire-dropzone](https://github.com/dasundev/livewire-dropzone) 
> - Turned off by drag and drop bug
> - Lacks slots and customizability

Made the font smaller on mobile size

Added bg-orange and text-orange class

Changes on vite configuration to make it accessible on phone 
Kindly check if the app still works fine when serving `php artisan serve --host 0.0.0.0` _localhost still works even host is 0.0.0.0_



https://github.com/user-attachments/assets/e9ebd0af-7f59-4170-9fd5-3ed2b9046eab
_Still finding the cause of why table overflow but not in this pr yet_
Accessing on phone
- visit the app through _IP address of machine_:8000 then the site address
- You may need to allow incoming requests on firewall inbound rule on the host machine to access the site on phone
- Enable usb or wireless debugging on phone
- You can inspect for errors in phone at chrome://inspect/#devices 

### Bug 
Intial bug 
![image](https://github.com/user-attachments/assets/55d94319-4fca-4999-9cfe-5b246073c63c)
https://drive.google.com/file/d/14iMK0QxIcyallyugfjIgBL-QR6UemJBc/view?usp=sharing

Fix but introduce another bug
https://drive.google.com/file/d/14m-1XhvuBz-yry9D0nMIfimtnaKHGiUq/view?usp=sharing

@Cedieziah I lost track what attempt  to fix I tried but here are attempts done that didn't fix the issue
- Adding a wire key manually inside the component
- Adding wire:ignore
- Using Modelable livewire attributes
- Try to remove the modal
- etc

C-Ivan-Bandilla commented:
I know why a lot of commit is in here 

the commits being like this if I don't have the branch in my local and pulled the branch from git

Im from staging then I create the branch  employee/hr/applications
then pulled employee/hr/applications from git

is this incorrect way?

are there other way to pull remote branch without causing this?

Carl-Tabuso commented:
The only reason I can think of is that you create a local branch with the name similar to the remote branch. But instead of adding the remote repo as an upstream, you pulled from it.

I don't think there's a "correct" way of branching if you're asking if this is incorrect.

Carl-Tabuso commented:
Can you have this fixed first? It's hard to identify the actual changes with buncha commits.

C-Ivan-Bandilla commented:
The error on job listing page 
Is due to this I overlooked this part as I didn't pull the changes, I just read the changes
![image](https://github.com/user-attachments/assets/c46835a9-4fd1-4a4d-907c-f90123cf311e)
is due to changes here https://github.com/POWERLANE-TEAM/emplo-hrms/commit/f7737537eac5996ceda80e47c35bb9c7f8215876#diff-38a3c817d2436f3515518e5ca157f26183c84b3f8c9d6b8ec8589041cc1121b1L30

the fix will be on different pr

I though the git messed up that causes the error lol

Carl-Tabuso commented:
I maybe made a slight typo in `JobVacancy` model method called `jobDetails` to `jobDetail`. You can search for files that get the job details from the child model `JobVacancy`. For example, this:

![image](https://github.com/user-attachments/assets/6ab6ab3d-b289-42db-b70c-2f13601ee7a0)

Carl-Tabuso commented:
Can you have @Cedieziah take a look here? I don't know what I should look and test here. One issue though, on `/preemploy` you have an n+1 issue:

![image](https://github.com/user-attachments/assets/a03ce6b9-e329-4cef-9035-0e89d95f8229)

Carl-Tabuso commented on app/Livewire/Employee/PreEmploymentModal.php:
can we create another subdirectory for modal components? Prob like this -> `Livewire/Employee/Modal/`

Carl-Tabuso commented:
Who's gonna implement this?

> Validation should include malware scan found free here https://github.com/sunspikes/clamav-validator

C-Ivan-Bandilla commented:
> Who's gonna implement this?
> 
> > Validation should include malware scan found free here https://github.com/sunspikes/clamav-validator

Might do this after moved to docker?

So the third party av is isolated from the personal pc

Carl-Tabuso commented:
I managed to remove duplicate queries on /preemploy. I thought a lot about the `AuthenticatedUserComposer` because this always ensures that a user is authenticated so there's no need to run another query on livewire components to check the user instance and `roles` relationship.

You can comment out the `mount()` method.

![image](https://github.com/user-attachments/assets/6231738c-7f74-46e6-9be0-067c63f239b0)

I also made a lil bit of changes to `AuthenticatedUserComposer`

from:
```php
        if ($guard->check()) {
            $user = Cache::flexible('user_' . $guard->id(), [30, 60], function () use ($guard) {
                return User::where('user_id', $guard->id())
                    ->with('roles')
                    ->first();
            });

            $role_name = $user->roles->pluck('name')->first();
            $view->with([
                'role_name' => $role_name,
                'user' => $user,
            ]);
        }
```

to:

```php
        if ($guard->check()) {
            $user = Cache::flexible('user_' . $guard->id(), [30, 60], function () use ($guard) {
                return $guard->user()->load('roles');
            });

            $role_name = $user->roles->pluck('name')->first();
            $view->with([
                'role_name' => $role_name,
                'user' => $user,
            ]); 
        }
```

The `$guard->check()` determines if the current user is authenticated, so there's no need to run a query of `User::where` , we can just get that currently authenticated user by `return $guard->user()`

This removes the duplicate queries on `model_has_roles` and `users` table.
![image](https://github.com/user-attachments/assets/192057dd-5a5a-4261-9015-02431401c8e3)

C-Ivan-Bandilla commented:
This is okay I already removed the role checking in both this and on employee side bar (not yet pushed)

C-Ivan-Bandilla commented:
I think I remember why I didnt use guard as my IDE or  the  PHP intellisense extnsions flags them as an error

![image](https://github.com/user-attachments/assets/15f0b3ba-0a17-4868-bf0c-35f711f69a0c)
![image](https://github.com/user-attachments/assets/5fa8a129-a572-4d7a-b05d-aa6eb81d5741)
But upon trying it works

Just documenting this

Carl-Tabuso commented:
About the ->load('roles'), i don't think you're going to need it anymore since most of the time we're checking by permissions.

C-Ivan-Bandilla commented:
1. I changed the application status to numeric and update its seeder 064cfab255682c874a2ec31c672d3acc9f663039
2. I added new permission for reemployment submission here I think having to delete his own document is different from able to update other document, wdyt about the permission? 0edec7719648a8dd855787bb59a0a16fc82aa834

- CREATE_PRE_EMPLOYMENT_DOCUMENT = 'create pre employment document'
- UPDATE_OWNED_PRE_EMPLOYMENT_DOCUMENT = 'update owned pre employment document'
- DELETE_OWNED_PRE_EMPLOYMENT_DOCUMENT = 'delete owned pre employment document'

3. I moved the applicant seeder from database seeder to the applicant seeder
4.  I moved the modals into livewire employee modals

Carl-Tabuso reviewed (approved):
I will keep these changes in mind. I went over the code and I didn't see anything that might cause issues.
